src/compiler/scala/tools/nsc/classpath/VirtualDirectoryClassPath.scala

```diff
@@ -30,6 +30,7 @@ case class VirtualDirectoryClassPath(dir: VirtualDirectory) extends ClassPath wi
     case Some(f) => dir.iterator.filter(f).toArray
     case _ => dir.toArray
   }
+  protected def hasChild(dir: AbstractFile, name: String): Boolean = dir.lookupName(name, directory = false).exists
```

Comment:
```suggestion
  protected def hasChild(dir: AbstractFile, name: String): Boolean = {
    val s = dir.lookupName(name, directory = false)
    s != null && s.exists
  }
```

This seems to fix the broken tests

Reply:
thanks! ill try

Reply:
looking at `lookupName` it always returns null for VirtualFile, there is probably another implementation needed.

Reply:
it seems that lookupName does the existence check as part of its contract so I will just check for non-null